roboflow/core/version.py

```diff
@@ -446,6 +446,46 @@ def live_plot(epochs, mAP, loss, title=""):
 
             time.sleep(5)
 
+        response = requests.get(f"{API_URL}/{workspace}/{project}/{self.version}?api_key={self.__api_key}")
+        if response.ok:
+            version_info = response.json()["version"]
+            has_model = bool(version_info.get("models"))
+        else:
+            has_model = False
+
+        if not has_model:
```

Comment:
I would cut this section, and try to instantiate a new `self.model` based on `self.type`, like you have below.

If we get here, we can assume there is a model, because we just ran a training.

The `has_model` logic here is based on another location, which had a bug that was recently fixed. Better to keep it simple and not do the extra request and logic.

Reply:
I did not have this check in my initial commit and added it based on PR review comments to make it more defensive. https://github.com/roboflow/roboflow-python/pull/325#discussion_r1784303524.  Anyways, It works either way.
@grzegorz-roboflow @SolomonLake , I removed it based on recent comments. please let me know final thoughts.

Reply:
Thanks @SolomonLake, yea - I'd normally go defensive in case saving training artifacts failed (I was not sure if at this point it is possible). I'm approving on my part.